core/src/sumeragi/message.rs

```diff
@@ -1,5 +1,7 @@
 //! Contains message structures for p2p communication during consensus.
 #![allow(clippy::module_name_repetitions)]
+#![allow(clippy::arithmetic, clippy::std_instead_of_core, clippy::std_instead_of_alloc)]
+#![allow(clippy::significant_drop_in_scrutinee)]
```

Comment:
1. `Significant drop` is just a _notification_ that somebody thought was a good idea to include in `restriction` rather than `nursery`. We might want  to revisit it once we have pruned the lints, but for now the benefits are not worth the effort. 

2. `arithmetic` is a lint that I intend to enable globally, once `cargo lints` is updated to ignore unknown lints. We already use `checked_add` and so forth in cases where it matters, everything else is going to add overhead and make the API more cumbersome. 

3. `std_instead_of_core` and `std_instead_of_alloc` as you said should be disabled globally. Using `core` is just a matter of changing the imports, while using `alloc` is much more involved, and in this case, completely unnecessary. While it's possible that our downstream users might be using Iroha in an embedded context I doubt that `no_std` can help if even a modest blockchain takes up Gigabytes of RAM.

Reply:
> While it's possible that our downstream users might be using Iroha in an embedded context I doubt that no_std can help if even a modest blockchain takes up Gigabytes of RAM.

No way they would host blockchain on an embedded device, but they might want to run `Iroha` client on embedded device at some point

Reply:
>  Iroha client on embedded device at some point

Certainly. At that point we should consider reducing the memory footprint by other means as well.